ResearchKit/Common/ORKAnswerFormat.m

```diff
@@ -1887,38 +1908,44 @@ - (instancetype)copyWithZone:(NSZone *)zone {
 - (BOOL)isAnswerValid:(id)answer {
     BOOL isValid = NO;
     if ([answer isKindOfClass:[NSString class]]) {
-        return [self isAnswerValidWithString:(NSString *)answer];
+        isValid = [self isAnswerValidWithString:(NSString *)answer];
     }
     return isValid;
 }
 
 - (BOOL)isAnswerValidWithString:(NSString *)text {
-    if (text.length > 0) {
-        return ([self isTextLengthValidWithString:text] && [self isEmailAddressValidWithString:text]);
+    BOOL isValid = YES;
+    if (text && text.length > 0) {
+        isValid = ([self isTextLengthValidWithString:text] && [self isTextRegexValidWithString:text]);
     }
-    
-    return YES;
+    return isValid;
 }
 
 - (BOOL)isTextLengthValidWithString:(NSString *)text {
     return (_maximumLength == 0 || text.length <= _maximumLength);
 }
 
-- (BOOL)isEmailAddressValidWithString:(NSString *)text {
-    if (self.isEmailAddress) {
-        NSPredicate *emailValidationTest = [NSPredicate predicateWithFormat:@"SELF MATCHES %@", EmailValidationRegex];
-        return [emailValidationTest evaluateWithObject:text];
+- (BOOL)isTextRegexValidWithString:(NSString *)text {
+    BOOL isValid = YES;
+    if (self.regex) {
+        NSString *regExPattern = self.regex;
+        NSRegularExpression *regEx = [[NSRegularExpression alloc] initWithPattern:regExPattern options:NSRegularExpressionCaseInsensitive error:nil];
```

Comment:
Consider to cache `NSRegularExpression` object as an ivar to avoid allocation during each time.